# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gnuchess/glibc.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./src/common.h.orig
+++ ./src/common.h
@@ -745,7 +745,7 @@ void wait_for_input(void);
  * Input routine, initialized to one of the specific
  * input routines. The given argument is the prompt.
  */
-void (*getline) (char *);
+void (*get_line) (char *);

 #define MAXSTR 128
 extern char inputstr[MAXSTR];
--- ./src/input.c.orig
+++ ./src/input.c
@@ -57,7 +57,7 @@ volatile int input_status = INPUT_NONE;
 char inputstr[MAXSTR];

 /*
- * The readline getline(), notice that even if we have readline,
+ * The readline get_line(), notice that even if we have readline,
  * it is not used if stdin is not a tty.
  */

@@ -127,7 +127,7 @@ void *input_func(void *arg __attribute__((unused)) )
              (RealGameCnt+1)/2 + 1 );
     }
     pthread_mutex_lock(&input_mutex);
-    getline(prompt);
+    get_line(prompt);
     input_status = INPUT_AVAILABLE;
     pthread_cond_signal(&input_cond);
     pthread_mutex_unlock(&input_mutex);
@@ -173,13 +173,13 @@ void InitInput(void)
 {
 #ifdef HAVE_LIBREADLINE
   if (isatty(STDIN_FILENO)) {
-    getline = getline_readline;
+    get_line = getline_readline;
     using_history();
   } else {
-    getline = getline_standard;
+    get_line = getline_standard;
   }
 #else
-  getline = getline_standard;
+  get_line = getline_standard;
 #endif
   /* Start input thread */
   pthread_create(&input_thread, NULL, input_func, NULL);
